src/server/pfs/server/driver.go

```diff
@@ -1972,76 +1935,122 @@ func (d *driver) deleteAll(ctx context.Context) error {
+					for i := record.OverwriteIndex.Index; int(i) < len(fileNode.FileNode.Objects); i++ {
+						delta -= sizeMap[fileNode.FileNode.Objects[i].Hash]
+					}
 				}
-				for _, record := range records.Records {
-					sizeMap[record.ObjectHash] = record.SizeBytes
-					if record.OverwriteIndex != nil {
-						// Computing size delta
-						delta := record.SizeBytes
-						fileNode, err := tree.Get(filePath)
-						if err == nil {
-							// If we can't find the file, that's fine.
-							for i := record.OverwriteIndex.Index; int(i) < len(fileNode.FileNode.Objects); i++ {
-								delta -= sizeMap[fileNode.FileNode.Objects[i].Hash]
-							}
-						}
 
-						if err := tree.PutFileOverwrite(filePath, []*pfs.Object{{Hash: record.ObjectHash}}, record.OverwriteIndex, delta); err != nil {
-							return err
-						}
-					} else {
-						if err := tree.PutFile(filePath, []*pfs.Object{{Hash: record.ObjectHash}}, record.SizeBytes); err != nil {
-							return err
-						}
-					}
+				if err := tree.PutFileOverwrite(filePath, []*pfs.Object{{Hash: record.ObjectHash}}, record.OverwriteIndex, delta); err != nil {
+					return err
 				}
 			} else {
-				nodes, err := tree.List(filePath)
-				if err != nil && hashtree.Code(err) != hashtree.PathNotFound {
+				if err := tree.PutFile(filePath, []*pfs.Object{{Hash: record.ObjectHash}}, record.SizeBytes); err != nil {
 					return err
 				}
-				var indexOffset int64
-				if len(nodes) > 0 {
-					indexOffset, err = strconv.ParseInt(path.Base(nodes[len(nodes)-1].Name), splitSuffixBase, splitSuffixWidth)
-					if err != nil {
-						return fmt.Errorf("error parsing filename %s as int, this likely means you're "+
-							"using split on a directory which contains other data that wasn't put with split",
-							path.Base(nodes[len(nodes)-1].Name))
-					}
-					indexOffset++ // start writing to the file after the last file
-				}
-				for i, record := range records.Records {
-					if err := tree.PutFile(path.Join(filePath, fmt.Sprintf(splitSuffixFmt, i+int(indexOffset))), []*pfs.Object{{Hash: record.ObjectHash}}, record.SizeBytes); err != nil {
-						return err
-					}
-				}
+			}
+		}
+	} else {
+		nodes, err := tree.List(filePath)
+		if err != nil && hashtree.Code(err) != hashtree.PathNotFound {
+			return err
+		}
+		var indexOffset int64
+		if len(nodes) > 0 {
+			indexOffset, err = strconv.ParseInt(path.Base(nodes[len(nodes)-1].Name), splitSuffixBase, splitSuffixWidth)
+			if err != nil {
+				return fmt.Errorf("error parsing filename %s as int, this likely means you're "+
+					"using split on a directory which contains other data that wasn't put with split",
+					path.Base(nodes[len(nodes)-1].Name))
+			}
+			indexOffset++ // start writing to the file after the last file
+		}
+		for i, record := range records.Records {
+			if err := tree.PutFile(path.Join(filePath, fmt.Sprintf(splitSuffixFmt, i+int(indexOffset))), []*pfs.Object{{Hash: record.ObjectHash}}, record.SizeBytes); err != nil {
+				return err
```

Comment:
The diff on this last part is a little bit hard to follow but I think it's almost all superficial changes that just come from the fact that you are now handling multiple write records and thus need to put the code in a loop... which leads to indentation... which leads to hairy diffs. Is that accurate?

Reply:
Close. If you look at line 2010, I'm actually removing a level of indentation. That's because a single `putFileRecords` can have a `tombstone` _and_ have data to be written (whereas before it was an if/else). But yea ... didn't really change anything in here except for the indentation.